doc.go

```diff
@@ -0,0 +1,83 @@
+// Generated by running
+//  go generate github.com/gonum/matrix
+// DO NOT EDIT.
+//
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// Package matrix provides common error handling mechanisms for matrix operations
+// in mat64 and cmat128.
+//
+// Overview
+//
+// This section provides a quick overview of the matrix package. The following
+// sections provide more in depth commentary.
+//
+// matrix provides:
+//  - Error type definitions
+//  - Error recovery mechanisms
+//  - Common constants used by mat64 and cmat128
+//
+// BLAS and LAPACK
```

Comment:
Do we really want this all here? matrix itself doesn't have these properties, though the errors relate to them. Maybe the matrix documentation should just say it provides error types and constants, and direct users to the documentation of mat64 and cmat128 for more explanation on how they can arise?